prefq/examples/imitation_preference_comparisons.py

```diff
@@ -172,52 +188,73 @@ def _wait_for_feedback_request(url):
                     env_make_kwargs={"render_mode": "rgb_array"},
 )
 
-reward_net = BasicRewardNet(
-    venv.observation_space, venv.action_space, normalize_input_layer=RunningNorm,
-)
-
-fragmenter = preference_comparisons.RandomFragmenter(warning_threshold=0, rng=rng)
-preference_model = preference_comparisons.PreferenceModel(reward_net)
-reward_trainer = preference_comparisons.BasicRewardTrainer(
-    preference_model=preference_model,
-    loss=preference_comparisons.CrossEntropyRewardLoss(),
-    epochs=3,
-    rng=rng,
-)
-
-agent = PPO(
-    policy=FeedForward32Policy,
-    policy_kwargs=dict(
-        features_extractor_class=NormalizeFeaturesExtractor,
-        features_extractor_kwargs=dict(normalize_class=RunningNorm),
-    ),
-    env=venv,
-    n_steps=2048 // venv.num_envs,
-    seed=0,
-)
-
-trajectory_generator = preference_comparisons.AgentTrainer(
-    algorithm=agent,
-    reward_fn=reward_net,
-    venv=venv,
-    exploration_frac=0.0,
-    rng=rng,
-)
-
-gatherer = PrefqGatherer(video_dir = video_dir, server_url=SERVER_URL)
-querent = preference_comparisons.PreferenceQuerent()
-
-pref_comparisons = preference_comparisons.PreferenceComparisons(
-    trajectory_generator,
-    reward_net,
-    num_iterations=5,
-    fragmenter=fragmenter,
-    preference_gatherer=gatherer,
-    reward_trainer=reward_trainer,
-    initial_epoch_multiplier=1,
-)
-
-pref_comparisons.train(total_timesteps=5_000, total_comparisons=200)
-
-reward, _ = evaluate_policy(agent.policy, venv, 10)
-print("Reward:", reward)
+class ContextManager():
```

Comment:
We should give this context manager a more descriptive name (`EnvClosingContext`).